packages/two_dimensional_scrollables/lib/src/table_view/table.dart

```diff
@@ -689,9 +691,11 @@ class RenderTableViewport extends RenderTwoDimensionalViewport {
           cell.layout(cellConstraints);
           cellParentData.layoutOffset = Offset(xPaintOffset, yPaintOffset);
         }
-        xPaintOffset += columnWidth;
+        xPaintOffset += columnWidth +
+            _columnMetrics[column]!.configuration.padding.trailing;
       }
-      yPaintOffset += rowHeight;
+      yPaintOffset +=
```

Comment:
Ah yes! Thank you, when I whipped this up I thought - wait it can't be this simple. Thanks for identifying that case. I will update and add a test that verifies we don't lay out unnecessary rows/columns on account of padding. :D